### PR TITLE
refactor: add documentation

### PR DESCRIPTION
**Pull request by ahv15**

### CODE REVIEW

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientParticleSystem.java [hunk 1]
```diff
@@ -35,6 +22,14 @@
     @In
     private EntityManager entityManager;
 
+    /**
+     * Attaches a particle emitter to the player when the player picks up a flag, which emits particles based on the flag's team.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientParticleSystem.java]:
```suggestion
     * Attaches a particle emitter to the player when the player picks up a flag.
     * Emitted particles are based on the flag's team.
```

File: src/main/java/org/terasology/ligthandshadow/componentsystem/events/ScoreUpdateFromServerEvent.java [hunk 2]
```diff
@@ -1,23 +1,13 @@
...
 package org.terasology.ligthandshadow.componentsystem.events;
 
 import org.terasology.engine.entitySystem.event.Event;
 import org.terasology.engine.network.BroadcastEvent;
 
+/**
+ * Notification event to indicate that the score has been updated.
+ */
 @BroadcastEvent
 public class ScoreUpdateFromServerEvent implements Event {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/events/ScoreUpdateFromServerEvent.java:12]: We might want to align the naming with the other notification events such as `OnFlagDropped`.
Same for the game state notification events.
As this PR is for adding documentation, I'm fine with doing so in a follow-up

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TakeBlockOnActivationSystem.java [hunk 3]
```diff
@@ -29,6 +16,9 @@
 import org.terasology.lightandshadowresources.components.TakeBlockOnActivateComponent;
 import org.terasology.ligthandshadow.componentsystem.events.GiveFlagEvent;
 
+/**
+ * System responsible for giving the flag only when a player from the opposing team tries to take it.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TakeBlockOnActivationSystem.java]:
```suggestion
 * System responsible for giving the flag to a player.
 * Players can only take flags of the opposing team.
```

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 4]
```diff
@@ -35,6 +38,12 @@
...
     private int blackScore;
 
+    /**
+     * Updates score when player tries to place the flag at their home base.
+     *
+     * @param event
+     * @param entity
+     */
     @ReceiveEvent(components = {WinConditionCheckOnActivateComponent.class, LASTeamComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java:48]: Personally, I'd like to rename this to `updateScoreOnActivate` (in a follow-up PR).
I always feel like event handlers' names should indicate what happens in reaction to the event, not only "repeat" the event that is handled...

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 5]
```diff
@@ -48,6 +57,14 @@ public void onRestartRequest(RestartRequestEvent event, EntityRef clientEntity,
         sendEventToClients(new ScoreUpdateFromServerEvent(LASUtils.BLACK_TEAM, blackScore));
     }
 
+    /**
+     * If the flag the player has corresponds to the opponent's team and the player places the flag at their home base
+     * the score of the player's team increases and the flag returns back to it's base.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]:
```suggestion
     * When a player places the opponent's flag at their home base, the player's team score
     * increases and the flag returns back to its team's base.
```

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 6]
```diff
@@ -24,6 +24,9 @@
 import org.terasology.lightandshadowresources.components.LASTeamComponent;
 import org.terasology.lightandshadowresources.components.WinConditionCheckOnActivateComponent;
 
+/**
+ * System responsible for calculating and providing score.
+ */
 @RegisterSystem(RegisterMode.AUTHORITY)
 public class ScoreSystem extends BaseComponentSystem {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java:31]: Looking at this system, I think that it is likely the next overhaul candidate.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/LASSystem.java [hunk 7]
```diff
@@ -45,7 +32,7 @@
     private CelestialSystem celestialSystem;
 
     /**
-     * Gives player inventory items on game start
+     * Gives an empty inventory to a player in the lobby to prevent fight's in the lobby.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/LASSystem.java:35]: > to prevent fights in the lobby

Actually, I think we might want to make players invulnerable while in the lobby and allow fights :thinking: 

However, as what you wrote matches the current state, this is rather a topic for a follow-up PR.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/GameEntitySystem.java [hunk 8]
```diff
@@ -16,6 +16,9 @@
 import org.terasology.ligthandshadow.componentsystem.components.LASConfigComponent;
 import org.terasology.engine.registry.In;
 
+/**
+ *  Provides an entity that keeps track of game state information.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/GameEntitySystem.java:20]: This is a system, not an entity, is it?

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/RestartSystem.java [hunk 9]
```diff
@@ -23,62 +10,36 @@
...
-                }
-                nuiManager.closeScreen(LASUtils.DEATH_SCREEN);
-            }
+        Iterable<EntityRef> clients = entityManager.getEntitiesWith(ClientComponent.class);
+        for (EntityRef client: clients) {
+            EntityRef player = client.getComponent(ClientComponent.class).character;
+            String team = player.getComponent(LASTeamComponent.class).team;
+            player.send(new RestoreFullHealthEvent(player));
+            player.send(new CharacterTeleportEvent(LASUtils.getTeleportDestination(team)));
+            client.send(new ClientRestartEvent());
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/RestartSystem.java]: This is not a documentation change. Please extract this to another PR.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java [hunk 10]
```diff
@@ -50,7 +37,7 @@
     private AssetManager assetManager;
 
     /**
-     * Change the Health HUD when the local player is spawned based on their Light and Shadow Team.
+     * Initializes the Health HUD when the local player is spawned based on the default white team.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java]:
```suggestion
     * Initializes the Health HUD when the local player is spawned in the lobby based on the default white team.
```

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java [hunk 11]
```diff
@@ -87,7 +74,7 @@ public void onCreateDefaultVisualCharacter(CreateVisualCharacterEvent event, Ent
     }
 
     /**
-     * Updates the skeletal mesh of a player when its team changes.
+     * Updates the skeletal mesh and the Health HUD of a player when the player teleports to their teams base and their team is changed.
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java]:
```suggestion
     * Updates the skeletal mesh and the Health HUD of a player when the player
     * chooses a team and teleports into the arena.
```

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientRestartSystem.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+                DeathScreen deathScreen = (DeathScreen) nuiManager.getScreen(LASUtils.DEATH_SCREEN);
+                MigLayout migLayout = deathScreen.find("playerStatistics", MigLayout.class);
+                if (migLayout != null) {
+                    migLayout.removeAllWidgets();
+                }
+                nuiManager.closeScreen(LASUtils.DEATH_SCREEN);
+            }
+        }
+    }
+}
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientRestartSystem.java]: This is not a documentation change, please extract it to another PR

**Review by jdrueckert — CHANGES_REQUESTED** [review]
I do have a few suggestions, but overall I like how you write documentation :+1:

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/GameEntitySystem.java (hunk 8 above)
Comment (ahv15) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/GameEntitySystem.java:20]: Yeah this is a system.

**Review by jdrueckert — APPROVED** [review]
